### PR TITLE
037: improve frontend error handling

### PR DESCRIPTION
**Pull request by ellamaolson**

### CODE REVIEW

File: client/src/app/user.service.ts [hunk 1]
```diff
@@ -33,22 +30,6 @@ export interface UserProfile {
...
-  prerequisites: [CourseData];
-  corequisites: [CourseData];
-  area: string;
-  type: number;
-  difficulty?: number;
-  impaction?: number;
-  credit: number;
-  termsOffered: string;
-}
-
```
Comment (ellamaolson) [inline client/src/app/user.service.ts:51]: CourseData belongs to the course service.

File: client/src/app/error-handler.service.ts [hunk 2]
```diff
@@ -42,53 +42,141 @@ export class ErrorHandlerService {
    * Match err to the error type and alert user
    * @param err
    */
-  findErrorType(errMsg): string {
-    console.log("This is the error: ", errMsg);
 
-    let resultString = "";
+  findErrorType(errMsg): string {
+    const errorsObj = {
```
Comment (DaleCS) [inline client/src/app/error-handler.service.ts:47]: awesome

File: client/src/app/user.service.ts [hunk 3]
```diff
@@ -152,7 +133,13 @@ export class UserService {
    * @param attribute (optional)
    */
   getProfile(attribute?: string) {
-    return this.http.get<UserProfile>(`${this.uri}/`, this.getHttpHeaders());
+    return this.http
+      .get<UserProfile>(`${this.uri}/`, this.getHttpHeaders())
+      .pipe(
+        catchError((err) => {
```
Comment (DaleCS) [inline client/src/app/user.service.ts:139]: I didn't know you could define catches before calling subscribe. Very cool. I was looking for something like this.

**Review by DaleCS — APPROVED** [review]
LGTM!

**Review by DaleCS — APPROVED** [review]
LGTM!

**Pull request merged**